src/TextButton/TextButton.js

```diff
@@ -113,17 +139,42 @@ export default class WrappedTextButton extends Component {
   }
 
   getAdditionalProps() {
-    let { type, shadow = true } = this.props
+    let { type = this.props.type, shadow = true } = this.getButtonState()
 
-    if (type === 'contained' && shadow) {
+    if (['contained', 'custom'].includes(type) && shadow) {
```

Comment:
nitpick; you could use something like `const RAISED_BUTTON_TYPES = new Set(['contained', 'custom'])` at the top of the file, then... 

```suggestion
    if (RAISED_BUTTON_TYPES.has(type) && shadow) {
```